logging/logger.go

```diff
@@ -46,7 +47,6 @@ type ZapCompatibleLogger interface {
 	Level() zapcore.Level
 	Named(name string) *zap.SugaredLogger
 	Sync() error
-	With(args ...interface{}) *zap.SugaredLogger
```

Comment:
I assume this means nothing in goutils was using this version of the With API?

And I see us only using the ZapCompatibleLogger for robot client connections. So removing the restrictions here seems totally safe. Just speaking out loud.

Reply:
> this means nothing in goutils was using this version of the With API

I believe there were a few usages of `With` from goutils, but judging from your comment on the goutils PR, maybe this note is resolved?

Reply:
Yep!